src/renderer/app/providers/context/NetworkContext/NetworkContext.tsx

```diff
@@ -59,45 +64,45 @@ export const NetworkProvider = ({ children }: PropsWithChildren) => {
     };
   }, [everyConnectionIsReady]);
 
-  const subscribeBalanceChanges = async (chain: ExtendedChain, accountIds: AccountId[]) => {
+  const subscribeBalanceChanges = (chain: ExtendedChain, accountIds: AccountId[]) => {
     if (!chain.api?.isConnected || !accountIds.length) return;
 
-    if (!hasSubscription(chain.chainId)) {
-      unsubscribe(chain.chainId);
-    }
-
     const relaychain = chain.parentId && connections[chain.parentId];
 
-    subscribe(chain.chainId, subscribeBalances(chain, relaychain, accountIds));
+    subscribe(chain.chainId, subscribeBalances(chain, accountIds, relaychain));
     subscribe(chain.chainId, subscribeLockBalances(chain, accountIds));
   };
 
-  const connectedConnections = Object.values(connections).filter(
-    (c) => c.connection.connectionStatus === ConnectionStatus.CONNECTED,
-  );
+  // subscribe to active accounts
+  useEffect(() => {
+    if (activeAccounts.length === 0) return;
 
-  const previousConnectedConnections = usePrevious(connectedConnections);
+    activeConnections.forEach((chain) => {
+      subscribeBalanceChanges(chain, accountUtils.getAllAccountIds(activeAccounts, chain.connection.chainId));
+    });
 
-  useEffect(() => {
-    if (activeAccounts.length === 0) {
+    return () => {
       unsubscribeAll();
+    };
+  }, [activeAccounts]);
 
-      return;
-    }
-
-    // subscribe to new connections
-    const newConnections = connectedConnections.filter((c) => !previousConnectedConnections?.includes(c));
+  // subscribe to new connections
+  useEffect(() => {
+    const newConnections = activeConnections.filter((c) => !prevConnections?.includes(c));
```

Comment:
This is O(N^2) check.  Not sure about impact on overall performance since the number of networks is less than 100 but me might use a Set to reduce time complexity here